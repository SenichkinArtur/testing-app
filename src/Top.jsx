import React from 'react';

function  render(items) {
    return items.map(
        (item, idx) => (
            <li key={idx}>{item}</li>
        )
    );
}

const Top = ({list = []}) => (
    <ul>{render(list)}</ul>
);

export default Top;
