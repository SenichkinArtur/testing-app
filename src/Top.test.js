import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Top from './Top';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('renders', () => {
    it(`render Top component without props`, () => {
      const component = shallow(<Top />);
      expect(component.find('ul')).toHaveLength(1);
      expect(component.find('li')).toHaveLength(0);
    });
   
    it(`render Top component with props`, () => {
      const component = shallow(<Top list={['item1', 'item2', 'item4']} />);
      expect(component.find('ul')).toHaveLength(1);
      expect(component.find('li')).toHaveLength(3);
    });
   
    it('match snapshot', () => {
      const component = shallow(<Top list ={ [1,2,3,4]} />);
      expect(component).toMatchSnapshot();
    });
});