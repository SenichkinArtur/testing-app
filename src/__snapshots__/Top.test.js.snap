// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders match snapshot 1`] = `ShallowWrapper {}`;
